examples/eval.rs

```diff
@@ -0,0 +1,59 @@
+#![allow(non_upper_case_globals, non_camel_case_types, non_snake_case, improper_ctypes)]
+
+//! # Running scripts
+//! Here is the under "Running scripts" in the MDN User Guide[1] translated into Rust. This only
+//! shows the ``run()`` function's contents because the original does as well.
+//!
+//! The actual code that is run is designed to be testable, unlike the example given.
+//! [1]: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_User_Guide
+//!
+
+
+use std::ptr;
+
+#[macro_use]
+extern crate mozjs;
+use mozjs::jsapi::*;
+use mozjs::rust::{JSEngine, RealmOptions, MutableHandle, Runtime};
+use mozjs::rust::SIMPLE_GLOBAL_CLASS;
+
+fn run(rt: Runtime) {
+    unsafe {
+        let options = RealmOptions::default();
+        rooted!(in(rt.cx()) let global =
+            JS_NewGlobalObject(rt.cx(), &SIMPLE_GLOBAL_CLASS, ptr::null_mut(),
+                               OnNewGlobalHookOption::FireOnNewGlobalHook,
+                               &*options));
+
+        /* These should indicate source location for diagnostics. */
+        let filename: &'static str = "inline.js";
+        let lineno: u32 = 1;
+
+        /*
+         * The return value comes back here -- if it could be a GC thing, you must
+         * add it to the GC's "root set" with the rooted! macro.
+         */
+        let mut rval: Value = std::mem::zeroed();
```

Comment:
We should be using the rooted! macro here to avoid unsafety.
```rust
rooted!(in(rt.cx()) let mut rval = UndefinedValue());
```
Then we'll use `rval.handle_mut()` later, and `rval.get()`.